src/main/kotlin/model/Bet.kt

```diff
@@ -0,0 +1,11 @@
+package model
+
+data class Bet(val amount: Double) {
+    companion object {
+        fun parseToBet(input: String): Bet {
+            val amount = input.trim().toDoubleOrNull() ?: throw IllegalArgumentException("Bet must be a valid number")
+            if (amount < 0) throw IllegalArgumentException("Amount should be greater than 0")
+            return Bet(amount)
+        }
+    }
+}
```

Comment:
I think it can change to value class 😉 
and how about move validation logics to constructor?
then it can get confidence with used constructor.

And, I wonder if handling string input or null values might be more appropriate in the view layer rather than in the model.

```suggestion
data class Bet(val amount: Double) {

    init {
        require(amount >= 0) { "Amount should be greater than 0" }
    }
    
    companion object {
        // ...
    }
}
```

Reply:
You're absolutely right — since Bet is a value class, it makes sense to lean into its strengths and ensure that all instances are valid by design. Moving the validation logic directly into the constructor (via init) would indeed make the class more self-contained and trustworthy, especially when used across different parts of the codebase.

That said, Kotlin's value class constructors can't be made private, so while we can enforce validation in the init block, we can't fully prevent misuse unless we wrap creation in a factory method. It's a trade-off between convenience and safety. What would you suggest?

As for handling string inputs and nulls — I completely agree that parsing and input validation (like checking for empty strings or malformed numbers) is better suited for the view layer.